### DISCUSSION TITLE
[Idea] Change the APE field validation to match XX.XXX format

### DISCUSSION THREAD

**Discussion by alexandrebak42** [PrestaShop/PrestaShop#34216] [Ideas]
### The problem

Hi,

In France the APE or NAF code is written either with the format XX.XXX or with the format XXXXX

The INSEE site which is the reference in this field writes it with XX.XXX

The problem is that the Validate::isApe function only validates the XXXXX format and the ape field in the PREFIX_customer table is a varchar(5).

### The solution

Change the Validate::isApe regex from **/^[0-9]{3,4}[a-zA-Z]{1}$/s** to **/^[0-9]{1,2}?.[0-9]{1,2}[a-zA-Z]{1}$/s** and change the ape field in the PREFIX_customer table to varchar(6).

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by MatShir** [comment]
Nice catch indeed -> https://entreprendre.service-public.fr/vosdroits/F33050?lang=en

**Reply by matks** [reply]
Suggested solution looks suitable 👍 just make sure to update all APE validation logic. This is one, I don't know if there are others (you might need to explore)

**Reply by boherm** [reply]
We have one more in `src/PrestaShopBundle/Form/Admin/Sell/Customer/CustomerType.php` at line 333.
Maybe we can create a new `ApeType` with a well good constraints already in it!

**Comment by boherm** [comment]
Fixed in https://github.com/PrestaShop/PrestaShop/pull/34242 by @alexandrebak42!
